src/lightning/pytorch/utilities/argparse.py

```diff
@@ -64,9 +64,9 @@ def insert_env_defaults(self: Any, *args: Any, **kwargs: Any) -> Any:
             kwargs.update(dict(zip(cls_arg_names, args)))
         env_variables = vars(_parse_env_variables(cls))
         # update the kwargs by env variables
-        kwargs = dict(list(env_variables.items()) + list(kwargs.items()))
+        kwargs = dict(list(kwargs.items()) + list(env_variables.items()))
```

Comment:
This is the fix. The env vars now take precendence over the arguments passed by the user.
This functionality is undocumented and so I am ok with changing that. I am confident that this was always the intention.